### PR TITLE
Confirmed assignments via assignment window

### PR DESCRIPTION
**Pull request by herdt-michael**
Introduce a way for the operator to confirm an assignment via the assignment window.

### CODE REVIEW

File: hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DeploymentManagementTest.java [hunk 1]
```diff
@@ -722,12 +732,69 @@ void assignDistributionSetToNotExistingTarget() {
         }
     }
 
+    @ParameterizedTest
```
Comment (ramannas) [inline hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DeploymentManagementTest.java:735]: Should we also add a test case to show how an action with "waiting_for_confirmation" status behaves when there is a new assignment? (simply gets canceled like the existing behavior)

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/UiProperties.java [hunk 2]
```diff
@@ -254,6 +254,11 @@ public static class Documentation implements Serializable {
              */
             private String maintenanceWindowView = "";
 
+            /**
+             * Link to documentation of the confirmation flow.
+             */
+            private String confirmationFlow = "";
+
```
Comment (ramannas) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/UiProperties.java:261]: it wont be just about the confirmation, but we need to mention about the user consent and then followed be the requiredConfirmation as individual assignments. I would suggest something like :
userConsentAndConfirmationGuide or userConsentAndConfirmationInfo

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/UiProperties.java [hunk 3]
```diff
@@ -353,6 +358,13 @@ public String getMaintenanceWindowView() {
                 return maintenanceWindowView;
             }
 
+            /**
+             * @return Link to documentation of the confirmation flow
+             */
+            public String getConfirmationFlow(){
```
Comment (ramannas) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/UiProperties.java:364]: naming could change here

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/UiProperties.java [hunk 4]
```diff
@@ -425,6 +437,10 @@ public void setMaintenanceWindowView(final String maintenanceWindowView) {
                 this.maintenanceWindowView = maintenanceWindowView;
             }
 
+            public void setConfirmationFlow(final String confirmationFlow) {
```
Comment (ramannas) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/UiProperties.java:440]: naming could change here

**Review by ramannas — CHANGES_REQUESTED** [review]
some minor comments

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/model/TargetWithActionType.java [hunk 5]
```diff
@@ -152,12 +154,21 @@ public String getMaintenanceWindowTimeZone() {
...
+    }
+
     @Override
     public String toString() {
         return "TargetWithActionType [controllerId=" + controllerId + ", actionType=" + getActionType() + ", forceTime="
                 + getForceTime() + ", weight=" + getWeight() + ", maintenanceSchedule=" + getMaintenanceSchedule()
                 + ", maintenanceWindowDuration=" + getMaintenanceWindowDuration() + ", maintenanceWindowTimeZone="
-                + getMaintenanceWindowTimeZone() + "]";
+                + getMaintenanceWindowTimeZone() + ", confirmationRequired=" + isConfirmationRequired() + "]";
     }
```
[the remark this answers is not in the document]
Comment (herdt-michael) [reply hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/model/TargetWithActionType.java:172]: adapted

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/UiProperties.java (hunk 2 above)
Comment (herdt-michael) [reply hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/UiProperties.java:261]: Changed to 'userConsentAndConfirmationGuide'

File: hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DeploymentManagementTest.java (hunk 1 above)
Comment (herdt-michael) [reply hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DeploymentManagementTest.java:736]: added

**Comment by herdt-michael** [comment]
Tackled all findings @ramannas. Please take another look at it.

**Review by ramannas — APPROVED** [review]
Thank a ton @herdt-michael  :) for the enhancement

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/model/DeploymentRequest.java [hunk 6]
```diff
@@ -48,18 +48,30 @@ public class DeploymentRequest {
...
-     *            cron expression is relative to this time zone.
+     *            example +02:00 for CET summer time and +00:00 for UTC. The start
+     *            time of a maintenance window calculated based on the cron
+     *            expression is relative to this time zone.
      *
+     * @param confirmationRequired
+     *            is a flag whether the confirmation should be required for the
+     *            resulting {@link Action} or not. In case the confirmation is not
+     *            required, the action will be automatically confirmed and putted in
+     *            the
```
[the remark this answers is not in the document]
Comment (herdt-michael) [reply hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/model/DeploymentRequest.java:59]: fixed

**Pull request merged**